scanner/src/main/kotlin/experimental/NestedProvenanceScanResult.kt

```diff
@@ -136,4 +137,75 @@ data class NestedProvenanceScanResult(
 
         throw IllegalArgumentException("Could not find entry for $provenance.")
     }
+
+    /**
+     * Remove all scan results for [nestedProvenance]s which are not within the provided [path] and filter all findings
+     * within the scan results by [path], taking the sub repository paths into account.
+     * This also updates the VCS paths for all [RepositoryProvenance]s accordingly to make clear that they are not
+     * results for the whole repository.
+     *
+     * Throws an [IllegalArgumentException] if any provenance in [nestedProvenance] already has a VCS path set.
+     */
+    fun filterByVcsPath(path: String): NestedProvenanceScanResult {
+        if (path.isBlank()) return this
+
+        val provenances = getProvenances()
+        val provenancesWithVcsPath = provenances.filter { it is RepositoryProvenance && it.vcsInfo.path.isNotBlank() }
+
+        require(provenancesWithVcsPath.isEmpty()) {
```

Comment:
Would it make sense to make this an invariant of `NestedProvenance` instead? e.g. move the check into that class?

Reply:
Maybe later at some point, but currently the `NestedProvenance` must be allowed, see my response to your next comment.